frontend/src/pages/demo/FormValidationDemo.tsx

```diff
@@ -0,0 +1,95 @@
+import { FunctionComponent } from 'react'
+import { useForm } from 'react-hook-form'
+import Button from '../../components/Button'
+import SelectField from '../../components/forms/SelectField'
+import TextField from '../../components/forms/TextField'
+import LayoutWithNav from '../../layouts/LayoutWithNav'
+
+enum Species {
+  Cat,
+  Dog,
+  Bird,
+}
+
+type Animal = {
+  name: string
+  age: number
+  species: Species
+  weight?: number
+}
+
+/**
+ * Demo of how to use `react-hook-form` with our custom components
+ */
```

Comment:
This was useful for me so I hope it can make this repo more approachable.

Reply:
ah this is definitely helpful, thank you!

Reply:
Suggestion (non-blocking): mention/link to demo code in the frontend README

Reply:
Great idea, done!